<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">
<html>
<head>
<title>Xapian: API Documentation: Xapian::ErrorHandler Class Reference</title>
<link href="doxygen.css" rel="stylesheet" type="text/css">
<link href="tabs.css" rel="stylesheet" type="text/css">
</head>
<body bgcolor="#ffffff">
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="inherits.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="namespaceXapian.html">Xapian</a></li><li class="navelem"><a class="el" href="classXapian_1_1ErrorHandler.html">ErrorHandler</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="classXapian_1_1ErrorHandler-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">Xapian::ErrorHandler Class Reference<span class="mlabels"><span class="mlabel">abstract</span></span></div>  </div>
</div><!--header-->
<div class="contents">

<p>Decide if a <a class="el" href="classXapian_1_1Error.html" title="All exceptions thrown by Xapian are subclasses of Xapian::Error. ">Xapian::Error</a> exception should be ignored.  
 <a href="classXapian_1_1ErrorHandler.html#details">More...</a></p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a2833c2fd77c9979574c63cee8db96e05"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a2833c2fd77c9979574c63cee8db96e05"></a>
&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classXapian_1_1ErrorHandler.html#a2833c2fd77c9979574c63cee8db96e05">ErrorHandler</a> ()</td></tr>
<tr class="memdesc:a2833c2fd77c9979574c63cee8db96e05"><td class="mdescLeft">&#160;</td><td class="mdescRight">Default constructor. <br/></td></tr>
<tr class="separator:a2833c2fd77c9979574c63cee8db96e05"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af4fccfeed11d85364088e5bed7f9048b"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="af4fccfeed11d85364088e5bed7f9048b"></a>
virtual&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classXapian_1_1ErrorHandler.html#af4fccfeed11d85364088e5bed7f9048b">~ErrorHandler</a> ()</td></tr>
<tr class="memdesc:af4fccfeed11d85364088e5bed7f9048b"><td class="mdescLeft">&#160;</td><td class="mdescRight">We require a virtual destructor because we have virtual methods. <br/></td></tr>
<tr class="separator:af4fccfeed11d85364088e5bed7f9048b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa3085bcf9a4e25bd2dd5d99710763b51"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classXapian_1_1ErrorHandler.html#aa3085bcf9a4e25bd2dd5d99710763b51">operator()</a> (<a class="el" href="classXapian_1_1Error.html">Xapian::Error</a> &amp;error)</td></tr>
<tr class="memdesc:aa3085bcf9a4e25bd2dd5d99710763b51"><td class="mdescLeft">&#160;</td><td class="mdescRight">Handle a <a class="el" href="classXapian_1_1Error.html" title="All exceptions thrown by Xapian are subclasses of Xapian::Error. ">Xapian::Error</a> object.  <a href="#aa3085bcf9a4e25bd2dd5d99710763b51">More...</a><br/></td></tr>
<tr class="separator:aa3085bcf9a4e25bd2dd5d99710763b51"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Decide if a <a class="el" href="classXapian_1_1Error.html" title="All exceptions thrown by Xapian are subclasses of Xapian::Error. ">Xapian::Error</a> exception should be ignored. </p>
<p>You can create your own subclass of this class and pass in an instance of it when you construct a <a class="el" href="classXapian_1_1Enquire.html" title="This class provides an interface to the information retrieval system for the purpose of searching...">Xapian::Enquire</a> object. <a class="el" href="classXapian_1_1Error.html" title="All exceptions thrown by Xapian are subclasses of Xapian::Error. ">Xapian::Error</a> exceptions which happen during the match process are passed to this object and it can decide whether they should propagate or whether <a class="el" href="classXapian_1_1Enquire.html" title="This class provides an interface to the information retrieval system for the purpose of searching...">Enquire</a> should attempt to continue.</p>
<p>The motivation is to allow searching over remote databases to handle a remote server which has died (both to allow results to be returned, and also so that such errors can be logged and dead servers temporarily removed from use). </p>
</div><h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="aa3085bcf9a4e25bd2dd5d99710763b51"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void Xapian::ErrorHandler::operator() </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classXapian_1_1Error.html">Xapian::Error</a> &amp;&#160;</td>
          <td class="paramname"><em>error</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Handle a <a class="el" href="classXapian_1_1Error.html" title="All exceptions thrown by Xapian are subclasses of Xapian::Error. ">Xapian::Error</a> object. </p>
<p>This method is called when a <a class="el" href="classXapian_1_1Error.html" title="All exceptions thrown by Xapian are subclasses of Xapian::Error. ">Xapian::Error</a> object is thrown and caught inside <a class="el" href="classXapian_1_1Enquire.html" title="This class provides an interface to the information retrieval system for the purpose of searching...">Enquire</a>. If this is the first <a class="el" href="classXapian_1_1ErrorHandler.html" title="Decide if a Xapian::Error exception should be ignored. ">ErrorHandler</a> that the <a class="el" href="classXapian_1_1Error.html" title="All exceptions thrown by Xapian are subclasses of Xapian::Error. ">Error</a> has been passed to, then the handle_error() virtual method is called, which allows the API user to decide how to handle the error.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">error</td><td>The <a class="el" href="classXapian_1_1Error.html" title="All exceptions thrown by Xapian are subclasses of Xapian::Error. ">Xapian::Error</a> object under consideration. </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>xapian/<a class="el" href="errorhandler_8h.html">errorhandler.h</a></li>
</ul>
</div><!-- contents -->
<hr>
<address><small>
Documentation for Xapian (version 1.2.17).<br>
Generated on Wed Jan 29 2014 by
<a href="http://www.doxygen.org/">Doxygen 1.8.5</a>.
</small></address>
</body>
</html>
